src/Service/Sqs/src/Result/ReceiveMessageResult.php

```diff
@@ -29,64 +30,94 @@ protected function populateResult(Response $response): void
-                    'MessageAttributes' => !$item->MessageAttribute ? [] : (function (\SimpleXMLElement $xml): array {
-                        $items = [];
-                        foreach ($xml as $item) {
-                            $items[$item->Name->__toString()] = !$item->Value ? null : new MessageAttributeValue([
-                                'StringValue' => ($v = $item->Value->StringValue) ? (string) $v : null,
-                                'BinaryValue' => ($v = $item->Value->BinaryValue) ? base64_decode((string) $v) : null,
-                                'StringListValues' => !$item->Value->StringListValue ? [] : (function (\SimpleXMLElement $xml): array {
-                                    $items = [];
-                                    foreach ($xml->StringListValue as $item) {
-                                        $a = ($v = $item) ? (string) $v : null;
-                                        if (null !== $a) {
-                                            $items[] = $a;
-                                        }
-                                    }
-
-                                    return $items;
-                                })($item->Value->StringListValue),
-                                'BinaryListValues' => !$item->Value->BinaryListValue ? [] : (function (\SimpleXMLElement $xml): array {
-                                    $items = [];
-                                    foreach ($xml->BinaryListValue as $item) {
-                                        $a = ($v = $item) ? base64_decode((string) $v) : null;
-                                        if (null !== $a) {
-                                            $items[] = $a;
-                                        }
-                                    }
-
-                                    return $items;
-                                })($item->Value->BinaryListValue),
-                                'DataType' => (string) $item->Value->DataType,
-                            ]);
-                        }
-
-                        return $items;
-                    })($item->MessageAttribute),
-                ]);
+        $this->Messages = !$data->Message ? [] : $this->populateResultMessageList($data->Message);
+    }
+
+    /**
+     * @return string[]
+     */
+    private function populateResultBinaryList(\SimpleXMLElement $xml): array
+    {
+        $items = [];
+        foreach ($xml->BinaryListValue as $item) {
+            $a = ($v = $item) ? base64_decode((string) $v) : null;
+            if (null !== $a) {
+                $items[] = $a;
+            }
+        }
+
+        return $items;
+    }
+
+    /**
+     * @return array<string, MessageAttributeValue>
+     */
+    private function populateResultMessageBodyAttributeMap(\SimpleXMLElement $xml): array
+    {
+        $items = [];
+        foreach ($xml as $item) {
+            $items[$item->Name->__toString()] = new MessageAttributeValue([
```

Comment:
previous code were `$items[$item->Name->__toString()] = !$item->Value ? null : new MessageAttributeValue([`

But this is not compatible with docbloc. Do youthink we should handle this (AS returns a map with null value)

Reply:
Yes. The previous code was more accurate.

Reply:
yes, but re-using the previous code is a BC break with documentation.
Currently maps are typehinted `array<string, Foo>` but the code my generate `array<string, null|Foo>`.

Hmm maybe we should just `continue` when value is null.

Reply:
pushed a better fix that handl null value and compatible with ou docblock